src/Bicep.Core/TypeSystem/TypeAssignmentVisitor.cs

```diff
@@ -303,27 +328,24 @@ private void ValidateDecorators(IEnumerable<DecoratorSyntax> decoratorSyntaxes,
                     TypeValidator.GetCompileTimeConstantViolation(argumentSyntax, diagnostics);
                 }
 
-                if (this.binder.GetSymbolInfo(decoratorSyntax.Expression) is FunctionSymbol symbol)
+                var symbol = this.binder.GetSymbolInfo(decoratorSyntax.Expression);
+
+                if (symbol is DeclaredSymbol or NamespaceSymbol)
                 {
-                    var argumentTypes = decoratorSyntax.Arguments
-                        .Select(argument =>
-                        {
-                            var argumentType = typeManager.GetTypeInfo(argument);
+                    diagnostics.Write(DiagnosticBuilder.ForPosition(decoratorSyntax.Expression).ExpressionNotCallable());

```

Comment:
>ExpressionNotCallable [](start = 96, length = 21)

If this error is only used here, should we make the message more explicit?

Reply:
Makes sense. I'll make it clearer that the expression cannot be used as a decorator.